src/full/Agda/Interaction/Options.hs

```diff
@@ -363,55 +330,24 @@ type Flag opts = opts -> OptM opts
 
 checkOpts :: Flag CommandLineOptions
 checkOpts opts
-  | htmlRelated = throwError htmlRelatedMessage
   | or [ p opts && matches ps > 1 | (p, ps) <- exclusive ] =
       throwError exclusiveMessage
   | otherwise = return opts
   where
   matches = length . filter ($ opts)
-  optionChanged opt = ((/=) `on` opt) opts defaultOptions
-
-  atMostOne =
-    [ optGenerateHTML
-    , isJust . optDependencyGraph
-    ] ++
-    map fst exclusive
+  atMostOne = map fst exclusive
 
   exclusive =
     [ ( optOnlyScopeChecking
       , optGenerateVimFile :
```

Comment:
A note on this: it would be nice to get the Vim syntax highlighting separated as a backend/plugin also. The reason I did not is that it's invoked eagerly during module type-checking; but backends are invoked _after_ all type-checking is complete. Turning it into a backend (under the current structure) would delay the generation of those files. I'm not sure if that's considered important or not.